pkg/doc/jose/jwk_test.go

```diff
@@ -0,0 +1,238 @@
+						}`,
+			},
+			{
+				name: "get public key bytes EC P-526 JWK",
+				jwkJSON: `{
+							"kty": "EC",
+							"use": "enc",
+							"crv": "P-256",
+							"kid": "sample@sample.id",
+							"x": "JR7nhI47w7bxrNkp7Xt1nbmozNn-RB2Q-PWi7KHT8J0",
+							"y": "iXmKtH0caOgB1vV0CQwinwK999qdDvrssKhdbiAz9OI",
+							"alg": "ES256"
+						}`,
+			},
+			{
+				name: "get public key bytes EC SECP256K1 JWK",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "rIJO8RmkExUecJ5i15L9OC7rl7pwmYFR8QQgdM1ERWI",
+   						 	"alg": "EdDSA"
+						}`,
+			},
+		}
+
+		t.Parallel()
+
+		for _, test := range tests {
+			tc := test
+			t.Run(tc.name, func(t *testing.T) {
+				pk, err := DecodePublicKey([]byte(tc.jwkJSON))
+				require.NoError(t, err)
+				require.NotEmpty(t, pk)
+			})
+		}
+	})
+
+	t.Run("Test decode public key failure", func(t *testing.T) {
+		tests := []struct {
+			name    string
+			jwkJSON string
+			err     string
+		}{
+			{
+				name:    "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `}`,
+				err:     "unable to read JWK",
+			},
+			{
+				name: "attempt public key bytes from invalid curve",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "sec12341",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "rIJO8RmkExUecJ5i15L9OC7rl7pwmYFR8QQgdM1ERWI",
+   						 	"alg": "EdDSA"
+						}`,
+				err: "unsupported elliptic curve 'sec12341'",
+			},
+			{
+				name: "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "",
+    						"y": "",
+   						 	"alg": "EdDSA"
+						}`,
+				err: "invalid EC public key, wrong length for x",
+			},
+			{
+				name: "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "",
+   						 	"alg": "EdDSA"
+						}`,
+				err: "invalid EC public key, wrong length for y",
+			},
+			{
+				name: "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "x",
+    						"y": "y",
+   						 	"alg": "EdDSA"
+						}`,
+				err: "unable to read JWK",
+			},
+			{
+				name: "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "{",
+    						"y": "y",
+   						 	"alg": "EdDSA"
```

Comment:
same here.. I think we need a mapping validation between `alg` and `crv` as they're related

Reply:
I would like to see a validation check between `crv` and `alg` that we currently support as follows:
ECDSA crv is valid only for P-256, P-384, P-521 and secp256k1 algs
EdDSA crv is valid only for Ed25519 alg

If we wanna support other key algs, we should add them to the list.

Reply:
I agree, go-jose library is doing that, I just branched out my logic when I see 'kty=EC,crv=secp256k1'. 'alg' is not mandatory field in jwk if crv/ec are explanatory enough.